tests/ImageSharp.Tests/Drawing/FillLinearGradientBrushTests.cs

```diff
@@ -0,0 +1,379 @@
+                    {
+                        // check first and last column:
+                        Assert.Equal(columnColor0, sourcePixels[0, i]);
+                        Assert.Equal(lastColumnColor, sourcePixels[lastColumnIndex, i]);
+
+                        // check the random colors:
+                        Assert.True(columnColor23.Equals(sourcePixels[23, i]), $"at {i}");
+                        Assert.Equal(columnColor42, sourcePixels[42, i]);
+                        Assert.Equal(columnColor333, sourcePixels[333, i]);
+                    }
+                }
+
+                image.CompareToReferenceOutput(provider);
+            }
+        }
+
+        [Theory]
+        [WithBlankImages(500, 10, PixelTypes.Rgba32, GradientRepetitionMode.DontFill)]
+        [WithBlankImages(500, 10, PixelTypes.Rgba32, GradientRepetitionMode.None)]
+        [WithBlankImages(500, 10, PixelTypes.Rgba32, GradientRepetitionMode.Repeat)]
+        [WithBlankImages(500, 10, PixelTypes.Rgba32, GradientRepetitionMode.Reflect)]
+        public void HorizontalGradientWithRepMode<TPixel>(
+            TestImageProvider<TPixel> provider,
+            GradientRepetitionMode repetitionMode)
+            where TPixel : struct, IPixel<TPixel>
+        {
+            using (var image = provider.GetImage())
+            {
+                int lastColumnIndex = image.Width - 1;
+
+                TPixel red = NamedColors<TPixel>.Red;
+                TPixel yellow = NamedColors<TPixel>.Yellow;
+
+                LinearGradientBrush<TPixel> unicolorLinearGradientBrush =
+                    new LinearGradientBrush<TPixel>(
+                        new SixLabors.Primitives.Point(0, 0),
+                        new SixLabors.Primitives.Point(image.Width / 10, 0),
+                        repetitionMode,
+                        new ColorStop<TPixel>(0, red),
+                        new ColorStop<TPixel>(1, yellow));
+
+                image.Mutate(x => x.Fill(unicolorLinearGradientBrush));
+                image.DebugSave(provider, repetitionMode);
+
+                using (PixelAccessor<TPixel> sourcePixels = image.Lock())
+                {
+                    TPixel columnColor0 = sourcePixels[0, 0];
+                    TPixel columnColor23 = sourcePixels[23, 0];
+                    TPixel columnColor42 = sourcePixels[42, 0];
+                    TPixel columnColor333 = sourcePixels[333, 0];
+
+                    TPixel lastColumnColor = sourcePixels[lastColumnIndex, 0];
+
+                    for (int i = 0; i < image.Height; i++)
+                    {
+                        // check first and last column:
+                        Assert.Equal(columnColor0, sourcePixels[0, i]);
+                        Assert.Equal(lastColumnColor, sourcePixels[lastColumnIndex, i]);
+
+                        // check the random colors:
+                        Assert.True(columnColor23.Equals(sourcePixels[23, i]), $"at {i}");
```

Comment:
Considering we have comparison against reference images now, do these assertions still add value to the tests?

Reply:
It depends.
If the reference images are checked with enough care, these don't add value.
If not, these checks do as these are some of the properties easy to calculate and check.

Nevertheless: Removing those to favor unit test performance over double-checking.